process_unix.cpp

```diff
@@ -15,6 +15,9 @@ Process::id_type Process::open(const std::string &command, const std::string &pa
     stderr_fd=std::unique_ptr<fd_type>(new fd_type);
   
   int stdin_p[2], stdout_p[2], stderr_p[2];
+  std::memset(stdin_p, 0, sizeof(stdin_p));
+  std::memset(stdout_p, 0, sizeof(stdout_p));
+  std::memset(stderr_p, 0, sizeof(stderr_p));
 
   if(stdin_fd && pipe(stdin_p)!=0) {
     close(stdin_p[0]);close(stdin_p[1]);
```

Comment:
When I think about it, when for instance `pipe(stdin_p)` fails, one should maybe not close `stdin_p[0]` and stdin_p[1]`. That is, I guess `stdin_p` is not set when `pipe(stdin_p)` fails. If so, the initialisation of `stdin_p` should not be needed.